compiler/noirc_evaluator/src/acir/tests/instructions.rs

```diff
@@ -1 +1,302 @@
+use acvm::assert_circuit_snapshot;
+
+use crate::acir::tests::ssa_to_acir_program;
+
 mod binary;
+
+#[test]
+fn constrain_equal() {
+    let src = "
+    acir(inline) fn main f0 {
+      b0(v0: Field, v1: Field):
+        constrain v0 == v1
+        return
+    }
+    ";
+    let program = ssa_to_acir_program(src);
+    assert_circuit_snapshot!(program, @r"
+    func 0
+    private parameters: [w0, w1]
+    public parameters: []
+    return values: []
+    ASSERT w1 = w0
+    ");
+}
+
+#[test]
+fn constrain_not_equal() {
+    let src = "
+    acir(inline) fn main f0 {
+      b0(v0: Field, v1: Field):
+        constrain v0 != v1
+        return
+    }
+    ";
+    let program = ssa_to_acir_program(src);
+
+    // `0 = w0*w2 - w1*w2 - 1` is `0 = (w0 - w1)*w2 - 1`, then `1 = (w0 - w1)*w2` and finally
+    // `w2 = 1/(w0 - w1)`. This means `w0 - w1` must not be zero, `w0 - w1 != 0` which is
+    // the same as `w0 != w1`.
+    assert_circuit_snapshot!(program, @r"
+    func 0
+    private parameters: [w0, w1]
+    public parameters: []
+    return values: []
+    BRILLIG CALL func: 0, inputs: [w0 - w1], outputs: [w2]
+    ASSERT 0 = w0*w2 - w1*w2 - 1
+
+    unconstrained func 0: directive_invert
+    0: @21 = const u32 1
+    1: @20 = const u32 0
+    2: @0 = calldata copy [@20; @21]
+    3: @2 = const field 0
+    4: @3 = field eq @0, @2
+    5: jump if @3 to 8
+    6: @1 = const field 1
+    7: @0 = field field_div @1, @0
+    8: stop &[@20; @21]
+    ");
+}
+
+#[test]
+fn cast() {
+    let src = "
+    acir(inline) fn main f0 {
+      b0(v0: u8):
+        v1 = cast v0 as Field
+        return v1
+    }
+    ";
+    let program = ssa_to_acir_program(src);
+
+    // cast is a no-op
+    // (casting in Noir might also involve a truncate operation if it's casting from
+    // a larger integer type to a smaller one, but casting in SSA is just changing types)
+    assert_circuit_snapshot!(program, @r"
+    func 0
+    private parameters: [w0]
+    public parameters: []
+    return values: [w1]
+    BLACKBOX::RANGE input: w0, bits: 8
+    ASSERT w1 = w0
+    ");
+}
+
+#[test]
+fn call_without_predicate() {
```

Comment:
Can be moved to the `call` test module and kind of looks like it is testing the same thing as some of the tests in that module